pkg/utils/utils_test.go

```diff
@@ -689,3 +690,35 @@ func TestIsValidCustomGPUArray(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckEnvsToRemove(t *testing.T) {
+	current := []v1.EnvVar{
+		{Name: "env1", Value: "value1"},
+		{Name: "env2", Value: "value2"},
+		{Name: "env3", Value: "value3"},
+		{Name: "env4", Value: "delete"},
+	}
+	desired := []v1.EnvVar{
+		{Name: "env2", Value: "value2"},
+		{Name: "env4", Value: "delete"},
+	}
+
+	needsToBeRemoved := []v1.EnvVar{
+		{Name: "env1", Value: "delete"},
```

Comment:
This `delete` should be `"env_marked_for_deletion"`, right?

Reply:
I think all `delete` value should be updated

Reply:
the delete value is this env is meant to be kept in this test.

Reply:
only envs explicitly removed from the isvc will be marked to deletion.

Reply:
Oops, yes. updated as it well, Thanks.